baremetal/metal3machine_manager.go

```diff
@@ -1052,7 +1052,7 @@ func (m *MachineManager) nodeReuseLabelExists(ctx context.Context, host *bmov1al
 
 // getBMCSecret will return the BMCSecret associated with BMH.
 func (m *MachineManager) getBMCSecret(ctx context.Context, host *bmov1alpha1.BareMetalHost) (*corev1.Secret, error) {
-	if host.Spec.BMC.CredentialsName == "" {
+	if host == nil || host.Spec.BMC.CredentialsName == "" {
```

Comment:
I believe this condition is already checked before you call this function no?

Reply:
Yse, from the current code, the `getBMCSecret` function is called after verifying that the host is not nil.
But I think `getBMCSecret` still needs this verification to avoid future calling it without verifying that the host is not nil.

Reply:
ok, for safety purpose, I agree
/lgtm